HIP-6.md

```diff
@@ -0,0 +1,147 @@
+| hip | title                   | status | author | created    |
+| --- | ----------------------- | ------ | ------ | ---------- |
+| 6   | Validator announcements | Draft  | asaj   | 2022-12-21 |
+
+### **Brief Summary / Abstract**
+
+Defines an optional specification for Hyperlane validators to announce themselves and the location of their signed checkpoints.
+
+### **Motivation**
+
+Hyperlane relayers need to aggregate validator signed checkpoints in order to deliver interchain messages.
+
+At present, relayers are manually configured with a list of validators and the location of their signed checkpoints.
+
+Relying on manual configuration creates friction to the introduction of additional validators via sovereign consensus.
+
+This HIP defines a standard for validators to announce themselves to relayers via a smart contract on the underlying blockchain.
+
+Relayers should query this contract to discover the location of validator signed checkpoints.
+
+### **Tech Spec**
+
+#### Announcment attestations
+
+The validator announcement protocol relies on validators signing attestations with the following semantic meaning:
+
+> "Checkpoints for the mailbox contract with address _m_ on chain _d_, signed by the validator with address _v_, can be found at location _l_
+
+The Hyperlane validator binary should sign the following data upon startup.
```

Comment:
IMO the binary should provide a separate command to create this attestation on demand?

Reply:
Why? Seems much more convenient to have it just show up without you needing to do anything

Reply:
I guess a separate commend feels more intuitive for me for a discrete output like this but I don't feel strongly. Having to SIGINT just to get that value seems unusual

Reply:
IMO sticking it in the storage modality under `announcement.json` is very convenient. If you're running a validator, it will be there, accessible to you, to register at any time.